File: frontend/src/types/recommendation.ts
export interface SetRecommendation {
  rank: number;
  set_id: number;
  api_set_id: string | null;
  set_name: string;
  series: string;
  release_date: string | null;
  missing_species_count: number;
  total_species_in_set: number;
  total_cards_in_set: number;
  coverage_percentage: number;
  missing_species_density_percentage: number;
}

export interface RecommendationResponse {
  total_species: number;
  owned_species: number;
  total_missing_species: number;
  recommendations: SetRecommendation[];
}

File: backend/app/schemas/recommendation.py
"""Response schemas for the pack recommendation engine."""

from datetime import date

from pydantic import BaseModel, ConfigDict

from app.schemas.progress import MissingSpeciesRead


class SetRecommendation(BaseModel):
    """A recommended set with its V1 score."""

    model_config = ConfigDict(from_attributes=True)

    set_id: int
    api_set_id: str | None
    set_name: str
    series: str
    release_date: date | None
    missing_species_count: int
    total_cards_in_set: int


class RecommendationResponse(BaseModel):
    """Top-level response for the recommendation endpoint."""

    total_missing_species: int
    recommendations: list[SetRecommendation]
